Automattic/pocket-casts-ios
# Patron: add subscription prices

Add Patron purchase flow.

## To test

### Pre-testing setup:
1. Open the Xcode Project
3. Hit <kbd>Command</kbd>+<kbd>Shift</kbd>+<kbd>,</kbd> to open the Scheme editor
4. Click the options tab
5. Select the `Pocket Casts Configuration.storekit` file in the StoreKit configuration dropdown

### Test the current flow

1. Make sure `patron` flag is disabled (Profile > Settings > Beta Features)
2. Test the existent flow and make sure your purchase is correctly done

### Test new flow
1. Make sure `patron` flag is enabled (Profile > Settings > Beta Features)
2. Log in with an account without Plus
3. Tap the folder icon under Podcasts
4. Change the price to monthly
5. Tap "Subscribe to Plus"
6. ✅ Monthly plus price should be selected
7. Dismiss modal
8. Change price to yearly
9. Tap "Subscribe to Plus"
10. ✅ Yearly plus price should be selected
11. Dismiss modal
12. Swipe to Patron
13. ✅ Perform the same tests (changing monthly/yearly and being reflected on the modal)
14. Change the Store Kit to have an introductory offer on any Patron plan (Go to the store kit file > select any Patron plan > Offer Type: Free)
15. Open the modal for Patron again (tap "Subscribe to Patron")
16. ✅ You should see the free offer displayed for the price
14. Buy any Patron plan
15. ✅ Make sure it works just fine

## Checklist
